File: benchmarks/tasks/fizzbuzz/fizzbuzz.cpp
// SPDX-License-Identifier: MIT-0

#include <iostream>
#include <string>

std::string fizzbuzz(int n) {
    if (n % 15 == 0) return "FizzBuzz";
    if (n % 3 == 0) return "Fizz";
    if (n % 5 == 0) return "Buzz";
    return std::to_string(n);
}

int main() {
    std::cout << fizzbuzz(15) << std::endl;
    return 0;
}

File: benchmarks/tasks/fizzbuzz/fizzbuzz.hs
-- SPDX-License-Identifier: MIT-0

fizzbuzz :: Int -> String
fizzbuzz n
  | n `mod` 15 == 0 = "FizzBuzz"
  | n `mod` 3  == 0 = "Fizz"
  | n `mod` 5  == 0 = "Buzz"
  | otherwise        = show n

main :: IO ()
main = putStrLn (fizzbuzz 15)
